### DISCUSSION TITLE
Hooks as function generator

### DISCUSSION THREAD

**Discussion by bernatvadell** [directus/directus#6961] [Feature Requests]
What would you think of having function generator as one more option?

I was thinking of something like this: 
```js
"items-hook.create": async *function({ ... }){
    // before
    yield;
   // after
}
```

**Comment by rijkvanzanten** [comment]
I've never really played too much with generator functions, but this definitely seems like an interesting use case!

**Reply by rijkvanzanten** [reply]
cc @nickrum

**Reply by nickrum** [reply]
Interesting idea! I'd be curious to hear what the advantages would be over the current approach.
Generator functions are mostly used to generate iterables with arbitrary length. In this case we will (likely) only ever have two cases to "generate". I feel like this might be confusing as you could add arbitrarily many `yield`s without a clear indication of what will be executed when and what will be executed at all.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
